Sources/NIOHTTP2/StreamStateMachine.swift

```diff
@@ -1023,3 +1022,34 @@ private extension HPACKHeaders {
         return self.first { $0.name == ":status" }?.value.first == "1"
     }
 }
+
+/// A state of a `HTTP2StreamStateMachine`. This copy mirrors `HTTP2StreamStateMachine.state` but without associated values.
+public enum NIOHTTP2StreamState {
```

Comment:
Ah, sorry, this should have occurred to me earlier: we need to make this an "extensible enum". To do that we actually need to define a wrapping struct, like this:

```swift
public struct NIOHTTP2StreamState: Hashable {
    private enum State {
        case idle
        case reservedRemote
        case reservedLocal
        case halfOpenLocalPeerIdle
        case halfOpenRemoteLocalIdle
        case fullyOpen
        case halfClosedLocalPeerIdle
        case halfClosedLocalPeerActive
        case halfClosedRemoteLocalIdle
        case halfClosedRemoteLocalActive
        case closed
    }

    private var state: State

    private init(state: State) {
        self.state = state
    }

    public static let idle = NIOHTTP2StreamState(state: .idle)
    public static let reservedRemote = NIOHTTP2StreamState(state: .reservedRemote)
    public static let reservedLocal = NIOHTTP2StreamState(state: .reservedLocal)
    public static let halfOpenLocalPeerIdle = NIOHTTP2StreamState(state: .halfOpenLocalPeerIdle)
    public static let halfOpenRemoteLocalIdle = NIOHTTP2StreamState(state: .halfOpenRemoteLocalIdle)
    public static let fullyOpen = NIOHTTP2StreamState(state: .fullyOpen)
    public static let halfClosedLocalPeerIdle = NIOHTTP2StreamState(state: .halfClosedLocalPeerIdle)
    public static let halfClosedLocalPeerActive = NIOHTTP2StreamState(state: .halfClosedLocalPeerActive)
    public static let halfClosedRemoteLocalIdle = NIOHTTP2StreamState(state: .halfClosedRemoteLocalIdle)
    public static let halfClosedRemoteLocalActive = NIOHTTP2StreamState(state: .halfClosedRemoteLocalActive)
    public static let closed = NIOHTTP2StreamState(state: .closed)
}
```

We can then keep the `fileprivate init` as-is.

The reasoning for doing this change is that in semantic versioning it is not possible to add new cases to an `enum` without shipping a new major version, because it's possible to exhaustively switch over enumerations. It is not possible to do so over `struct`s, so having a `struct` backed by an `enum` is an easy way to get a similar programming interface without the exhaustive switch support.

Reply:
I will work on this change.

Reply:
Done.